multiqc/plots/beeswarm.py

```diff
@@ -33,9 +35,10 @@ def plot (data, headers=None, pconfig=None):
 
 def make_plot(dt):
 
-    bs_id = dt.pconfig.get('id', 'table_{}'.format(''.join(random.sample(letters, 4))) )
+    bs_id = dt.pconfig.get( 'id', 'table_{}'.format(id(dt)) )
 
     # Sanitise plot ID and check for duplicates
+    from multiqc.utils import report
```

Comment:
Is this causing some kind of recursion error or something?

Reply:
Yes it's more of me trying to avoid circular imports and global variables in order to maintain loose-coupling and hence enable unit testing.
Ideally, IMHO, the plot libraries should not need to see the global report object in order to generate HTML snippets. When I started the fork, they could be decoupled with the count_hc_plots()/count_mpl_plots() functions that I added in place of global counters, but since then there is more coupling added, so I moved the imports as a compromise.

Reply:
Yes, there's probably a better way of doing it. I hesitated before writing it that way, but I needed the global `report.html_ids` list to keep track of the IDs as they were being created to avoid duplicates and I couldn't think of an alternative way to do that.